app/models/event.rb

```diff
@@ -6,4 +6,8 @@ class Event
   field :periodicity, type: Integer, default: 1
   belongs_to :owner, class_name: 'User'
   has_many :invitations, dependent: :destroy
+
+  validates_presence_of :name, :date
+  validates_presence_of :owner, on: :create
+  validates_uniqueness_of :name, scope: :owner
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.